crypto/cryptlib.c

```diff
@@ -355,4 +355,54 @@ int CRYPTO_memcmp(const void * in_a, const void * in_b, size_t len)
 
     return x;
 }
+
+/*
+ * For systems that don't provide an instruction counter register or equivalent
+ * we provide a best available clock.
+ * Return zero if no such timer is available.
+ */
+uint32_t OPENSSL_rdtsc(void)
+{
+# if defined(_AIX)
+    timebasestruct_t t;
+
+    read_wall_time(&t, TIMEBASE_SZ);
+    if (time_base_to_time(&t, TIMEBASE_SZ) == 0)
+        return (uint32_t)(t.tb_high ^ t.tb_low);
+    return 0;
+# elif defined(__sun) || defined(SOLARIS) || defined(__hpux)
+    return (uint32_t)gethrtime();
+# elif defined(_WIN32)
+    LARGE_INTEGER t;
+
+    QueryPerformanceCounter(&t);
+    return (uint32_t)t.QuadPart;
+# elif defined(_POSIX_C_SOURCE) \
+       && defined(_POSIX_TIMERS) \
+       && _POSIX_C_SOURCE >= 199309L
```

Comment:
Oh, I see: I should have read the complete manpage to learn about `_POSIX_TIMERS`. Are you sure these guards will guarantee that `clock_gettime` will be available without `-lrt`? Otherwise we spin around back to the [original problem](https://mta.openssl.org/pipermail/openssl-commits/2018-January/017804.html) where we started in #5199.

Reply:
My impression is that any usage of clock_gettime() will require -lrt on linux, at least on with older glibc version.

Reply:
`-lrt` will still be required on older Linux systems.  The original problem hasn't been solved by this.  That's why I tagged it for 1.2.  It would, however, be possible to exclude the `clock_gettime` code path for platforms that might need `-lrt`.  The fall back would then be to `gettimeofday`.

Reply:
Adding `&& (!defined(__GLIBC__) || __GLIBC_PREREQ(2, 17))` and including <features.h> should be enough to avoid needing to link against `-lrt`.  It's a bit hacky but what do people think?